### PR TITLE
Adds SASL_SSL support

### PR DESCRIPTION
**Pull request by abohmeed**
**What this PR does / why we need it**:
Adds SASL with the option to select the authentication mechaism
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @abohmeed. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: executor/api/kafka/server.go
Comment (agrski) [inline executor/api/kafka/server.go:209]: :thought_balloon: The username and password options apply for both `SASL_SSL` *and also* `SASL_PLAINTEXT`, so we should really support both here, as it's basically no extra effort.

Comment (agrski) [reply executor/api/kafka/server.go:209]: If we do that, this conditional should not be nested but rather another top-level conditional.

File: executor/api/kafka/server.go
Comment (agrski) [inline executor/api/kafka/server.go:108]: :question: Does using files instead of strings work for username and password?  I can't see how the underlying Kafka library would differentiate the intents, but happy to be pointed to docs!

**Review by agrski — CHANGES_REQUESTED** [review]
Thanks for putting in a PR to improve the Kafka support in the executor!

The current state looks to make a good start, but will need a few changes.

**Comment by abohmeed** [comment]
I'll reopen

**Pull request closed without merging**
